algorithms/linked_list.py

```diff
@@ -1,57 +1,69 @@
-class Node():
-	def __init__(self, value, next):
-		self.value=value
-		self.next=next
+class Node:
+	def __init__(self, value, next = None):
+		self.value = value
+		self.next = next
 
-class LinkedList():
-	def __init__(self, head, tail):
-		self.head=head
-		self.tail=tail
+	def __repr__(self):
+		return "Node(value={}, next={})".format(self.value, self.next)
 
-	def add(self, node):
-		self.tail.next=node
-		self.tail=node
+class LinkedList:
+	def __init__(self, head):
+		self.head = head
+		self.tail = self.set_tail()
+
+	def __repr__(self):
+		return "LinkedList(head={}, tail={})".format(self.head, self.tail)
+
+	def set_tail(self):
+		# finds the last node in the head
+		current_node = self.head
+		while current_node and current_node.next:
+			current_node = current_node.next
+		return current_node
+
+	def add(self, new_node):
+		if self.tail:
```

Comment:
If a list is size one, it's single node is both the head and the tail.

If the list did not have a tail, then it would mean the list is empty.  THen the newly added node would be both the head and the tail